internal/plugin/data_plane_server.go

```diff
@@ -29,23 +30,29 @@ type (
 	}
 
 	DataPlaneServer struct {
-		address      string
-		logger       *slog.Logger
-		instances    []*instances.Instance
-		configEvents map[string][]*instances.ConfigurationStatus
-		messagePipe  bus.MessagePipeInterface
-		server       *http.Server
-		cancel       context.CancelFunc
+		address           string
+		logger            *slog.Logger
+		instances         []*instances.Instance
+		configEvents      map[string][]*instances.ConfigurationStatus
+		messagePipe       bus.MessagePipeInterface
+		server            *http.Server
+		cancel            context.CancelFunc
+		serverMutex       *sync.Mutex
+		configEventsMutex *sync.Mutex
+		instancesMutex    *sync.Mutex
 	}
 )
 
 func NewDataPlaneServer(agentConfig *config.Config, logger *slog.Logger) *DataPlaneServer {
 	address := net.JoinHostPort(agentConfig.DataPlaneAPI.Host, strconv.Itoa(agentConfig.DataPlaneAPI.Port))
 
 	return &DataPlaneServer{
-		address:      address,
-		logger:       logger,
-		configEvents: make(map[string][]*instances.ConfigurationStatus),
+		address:           address,
+		logger:            logger,
+		configEvents:      make(map[string][]*instances.ConfigurationStatus),
+		serverMutex:       &sync.Mutex{},
+		configEventsMutex: &sync.Mutex{},
```

Comment:
What's the distribution of reads and writes of config events? Would we expect more reads than writes, or is it roughly equal?

Reply:
As discussed on Slack, we will default to `sync.Mutex` unless we notice a need for a `sync.RWMutex` (when it is obvious that there are far more reads than writes).